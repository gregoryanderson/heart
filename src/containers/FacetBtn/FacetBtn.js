import "./FacetBtn.css";
import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";


export const FacetBtn = props => {

  // const color = {assignment}

  
  const colorStyle = {
    backgroundColor: props.assignment
  };
  
  return (
    <Link to={`/${props.name}/${props.assignment}`}>
      <button style={colorStyle} onClick={() => props.handleSearch(props.name, props.assignment)}>
        {props.assignment}
      </button>
    </Link>
  );
};

export default FacetBtn;
