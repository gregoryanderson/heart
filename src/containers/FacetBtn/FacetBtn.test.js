import React from "react";
import { shallow } from "enzyme";
import { FacetBtn } from "./FacetBtn";

describe("FacetBtn", () => {
  let wrapper, mockHandleSearch;

  beforeEach(() => {
    mockHandleSearch = jest.fn();

    wrapper = shallow(
      <FacetBtn
        assignment='artist'
        key={123}
        name='Van Gogh'
        handleSearch={mockHandleSearch}
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleSearch when clicked", () => {
    wrapper.find("button").simulate("click");

    expect(mockHandleSearch).toHaveBeenCalled();
  });
});
