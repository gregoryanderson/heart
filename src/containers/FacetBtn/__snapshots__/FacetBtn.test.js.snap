// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacetBtn should match the snapshot 1`] = `
<Link
  to="/Van Gogh/artist"
>
  <button
    onClick={[Function]}
  >
    artist
  </button>
</Link>
`;
